provisioning/device/src/dps_errors.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+import { errors } from 'azure-iot-common';
+
+/**
+ * @private
+ */
+export class DpsError extends Error {
```

Comment:
>DpsError [](start = 13, length = 8)

ProvisioningError?
Also, response and responseBody are very HTTP-ish in the SDK - I'd be curious to see how that works out for other transports.

FWIW we may just get away with a single `transportObj` property like we do in other parts of the IoT Hub SDK. and maybe even make it private so that users don't take a dependency on it (or at least make it clear it's a debug thing that may change)